petab/lint.py

```diff
@@ -65,6 +93,79 @@ def condition_table_is_parameter_free(condition_df):
             return False
     return True
 
+def check_parameter_sheet(problem):
+    check_parameter_df(problem.parameter_df)
+    parameterId_is_string(problem.parameter_df)
+    parameterScale_is_valid(problem.parameter_df)
+    parameterBounds_are_numeric(problem.parameter_df)
+    parameterEstimate_is_boolean(problem.parameter_df)
+    parameterId_is_unique(problem.parameter_df)
+    check_parameterBounds(problem.parameter_df)
+    return True
+
+def parameterId_is_string(parameter_df):
+    """Check if all entries in the parameterId column of the parameter table are string and not empty"""
+
+    for parameterId in parameter_df['parameterId']:
+        if isinstance(parameterId, str):
+            if parameterId[0].isdigit():
+                raise ValueError('parameterId ' + parameterId + ' starts with integer')
+        else:
+            raise ValueError('Empty parameterId found')
+
+    return True
+
+def parameterId_is_unique(parameter_df):
+    """Check if the parameterId column of the parameter table is unique"""
+    if len(parameter_df['parameterId']) == len(set(parameter_df['parameterId'])):
+        return True
+    else:
+        raise AssertionError('parameterId column in parameter table is not unique')
+
+
+def parameterScale_is_valid(parameter_df):
+    """Check if all entries in the parameterScale column of the parameter table are
+    'lin' for linear, 'log' for natural logarithm or 'log10' for base 10 logarithm """
+
+    for parameterScale in parameter_df['parameterScale']:
+        if parameterScale not in ['lin', 'log', 'log10']:
+            raise AssertionError('Expected "lin", "log" or "log10" but got "'+parameterScale+'"')
+
+    return True
+
+def parameterBounds_are_numeric(parameter_df):
+    """Check if all entries in the lowerBound and upperBound column of the parameter table are
+    numeric """
+
+    for lowBound in parameter_df['lowerBound']:
```

Comment:
Pandas will automatically provide a float-type column if all values match the format. So instead of looping across the whole list, just check `Series.dtype`